sql/sql_expression_operations.cc

```diff
@@ -2851,6 +2851,105 @@ getChildren() const
     throw HttpReturnException(500, "Unknown IN expression type");
 }
 
+/*****************************************************************************/
+/* LIKE EXPRESSION                                                           */
+/*****************************************************************************/
+
+LikeExpression::
+LikeExpression(std::shared_ptr<SqlExpression> left,
+               std::shared_ptr<SqlExpression> right,
+               bool negative)
+    : left(std::move(left)),
+      right(std::move(right)),
+      isnegative(negative)
+{
+}
+
+BoundSqlExpression
+LikeExpression::
+bind(SqlBindingScope & context) const
+{
+    BoundSqlExpression boundLeft  = left->bind(context);
+    BoundSqlExpression boundRight  = right->bind(context);
+
+    return {[=] (const SqlRowScope & rowScope,
+                     ExpressionValue & storage,
+                     const VariableFilter & filter) -> const ExpressionValue &
+        {
+            ExpressionValue vstorage, fstorage;
+
+            const ExpressionValue & value = boundLeft(rowScope, vstorage, filter);
+
+            if (!value.isString())
+                throw HttpReturnException(400, "LIKE expression expected its left hand value to be a string, got " + value.getTypeAsUtf8String());
+
+            const ExpressionValue & filterEV = boundRight(rowScope, fstorage, filter);
```

Comment:
The filter is not necessarily constant. You could build it or take in in a dataset.